lib/train/errors.rb

```diff
@@ -40,4 +40,7 @@ class PlatformDetectionFailed < Error; end
 
   # Exception for when a invalid cache type is passed.
   class UnknownCacheType < Error; end
+
+  # Exception for when a command reaches configured timeout
```

Comment:
This _maybe_ should not be here, since it's only used by ssh_connection, and that's technically a plugin. I don't feel it's wrong to have a set of errors in core train that are standard for plugins though.

Reply:
I definitely want to implement this for winrm as well